syntax = "proto3";

package kernel_gym;

message Int64List {
  repeated int64 value = 1;
}

message ScalarRange {
  ScalarLimit min = 1;
  ScalarLimit max = 2;
}

message ScalarLimit {
  double value = 1;
}

message ScalarRangeList {
  repeated ScalarRange range = 1;
}


message Observation {
  oneof value {
    Int64List int64_list = 1;
    int64 scalar_int64 = 2;
  }
}

message ObservationSpace {
  string name = 1;
  oneof shape {
    ScalarRangeList int64_range_list = 2;
    ScalarRange scalar_int64_range = 3;
  }
  Observation default_value = 4;
}